### ISSUE TITLE
[icu] build on Mac required autoconf,automake,autoconf-archive

### ISSUE THREAD

**Issue by drink-crow** [microsoft/vcpkg#24870]
I try build icu on macOS with Apple M1 and it failuer. I check the err log, and found that required autoconf,automake,autoconf-archive. With installing them by brew, `./vcpkg install icu:arm64-osx` success.
this envirement should be checked in build tree

**Host Environment**
- OS: macOS
- Compiler: clang

**Comment by JackBoosY** [comment]
Can you please provide the failure logs?

**Comment by drink-crow** [comment]
without autoconf,automake,autoconf-archive, first it show that
```
CMake Error at scripts/cmake/vcpkg_configure_make.cmake:619 (message):
  icu requires autoconf from the system package manager (example: "sudo
  apt-get install autoconf")
Call Stack (most recent call first):
  ports/icu/portfile.cmake:53 (vcpkg_configure_make)
  scripts/ports.cmake:146 (include)
```
so i install autoconf, and then
```
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:128 (message):
    Command failed: /opt/homebrew/bin/autoreconf -vfi
    Working Directory: /Users/Mink/Code/vcpkg/buildtrees/icu/src/c-71_1-src-08c83f7de1.clean/source
    Error code: 2
    See logs for more information:
      /Users/Mink/Code/vcpkg/buildtrees/icu/autoconf-x64-osx-err.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_configure_make.cmake:629 (vcpkg_execute_required_process)
  ports/icu/portfile.cmake:53 (vcpkg_configure_make)
  scripts/ports.cmake:146 (include)
```
i checked `/Users/Mink/Code/vcpkg/buildtrees/icu/autoconf-x64-osx-err.log`, it say
```
autoreconf: export WARNINGS=
autoreconf: Entering directory '.'
autoreconf: configure.ac: not using Gettext
autoreconf: running: aclocal --force 
Can't exec "aclocal": No such file or directory at /opt/homebrew/Cellar/autoconf/2.71/share/autoconf/Autom4te/FileUtils.pm line 274.
autoreconf: error: aclocal failed with exit status: 2
```
i search `aclocal` and then install `automake`, and then install more again, it say
```
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:128 (message):
    Command failed: /bin/bash -c "V=1 ./../src/c-71_1-src-08c83f7de1.clean/source/configure --host=x86_64-apple-darwin \"--disable-samples\" \"--disable-tests\" \"--disable-layoutex\" \"--with-cross-build=/Users/Mink/Code/vcpkg/installed/arm64-osx/tools/icu\" \"--disable-silent-rules\" \"--verbose\" \"--disable-shared\" \"--enable-static\" \"--enable-debug\" \"--disable-release\" \"--prefix=/Users/Mink/Code/vcpkg/installed/x64-osx/debug\" \"--bindir=\\${prefix}/../tools/icu/debug/bin\" \"--sbindir=\\${prefix}/../tools/icu/debug/sbin\" \"--libdir=\\${prefix}/lib\" \"--includedir=\\${prefix}/../include\" \"--datarootdir=\\${prefix}/share/icu\""
    Working Directory: /Users/Mink/Code/vcpkg/buildtrees/icu/x64-osx-dbg
    Error code: 1
    See logs for more information:
      /Users/Mink/Code/vcpkg/buildtrees/icu/config-x64-osx-dbg-out.log
      /Users/Mink/Code/vcpkg/buildtrees/icu/config-x64-osx-dbg-err.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_configure_make.cmake:811 (vcpkg_execute_required_process)
  ports/icu/portfile.cmake:53 (vcpkg_configure_make)
  scripts/ports.cmake:146 (include)
```
check `/Users/Mink/Code/vcpkg/buildtrees/icu/config-x64-osx-dbg-err.log`, it say
```
configure: WARNING: unrecognized options: --disable-silent-rules
configure: error: 'autoconf-archive' is missing
```
install `autoconf-archive`, finally, success.

**Comment by JackBoosY** [comment]
cc @Neumann-A @dg0yt If you guys have any PRs to fix `vcpkg_configure_make`, I hope you improve the error message according to this issue.
Thank you very much.

**Comment by dg0yt** [comment]
@JackBoosY As you see, there already is a request for `autoconf` from `scripts/cmake/vcpkg_configure_make.cmake`. Does it always need `automake`, too? (I don't know). I'm sure not all ports need `autoconf-archive`, but this may change over time.
(I'm not a fan of always running autoreconf. An up-to-date release tarball doesn't need it, and it is slow on Windows.)

**Comment by JackBoosY** [comment]
@dg0yt We received many issue about that:
https://github.com/microsoft/vcpkg/issues/24806

**Comment by Neumann-A** [comment]
What we need is the script version of the make based stuff (e.g. `vcpkg_make_configure`) and then metaports to declare those implicit dependencies and make them explicit. 

>  An up-to-date release tarball doesn't need it

depends. Often the tarballs are packed with an old version of libtool. Rerunning autoreconf always makes sure that the scripts are all according to what vcpkg expects. Also, running autoreconf is often quite fast compared to configure itself.

**Comment by bavulapati** [comment]
This is still valid. And it also happens on OSX x64

**Comment by dg0yt** [comment]
Resolved by #38019.
